src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1899,7 +1890,7 @@ private void EnterParameters()
             }
 
             var methodParameters = methodSymbol.Parameters;
-            var signatureParameters = (_useDelegateInvokeParameterTypes ? _delegateInvokeMethod : methodSymbol).Parameters;
+            var signatureParameters = (_useDelegateInvokeParameterTypes ? _delegateInvokeMethod! : methodSymbol).Parameters;
```

Comment:
I would add a corresponding assertion since we don't have a check in the method to justify the suppression

Reply:
The dereference will result in an exception if this is incorrect.

---
In reply to: [463268294](https://github.com/dotnet/roslyn/pull/45640#discussion_r463268294) [](ancestors = 463268294)